### PR TITLE
Introduce sly-connection-update-interval

### LINKED ISSUE #378: Make (all?) timers configurable (in particular: instant startup!)
I just found out that if I set the `sly-attempt-connection` timer from 0.3 to 0.1, I get near-instant REPL startup (using an optimized core).  

It's a dealbreaker in my opinion for those among us who use SLY as their main shell ;)

So I suggest making this a defcustom.  I wonder if changing the default value to 0.2 or 0.1 wouldn't be better.

Another timer is in `sly-db-exit`.

We already has the `sly-threads-update-interval` timer defcustom.
So I suggest adding `sly-exit-update-interval` and `sly-connection-update-interval`.

Thoughts?

### PR DESCRIPTION
**Pull request by Ambrevar**
Fixes #378.

### CODE REVIEW

**Comment by joaotavora** [comment]
`sly-connection-poll-interval` is probably a better name.  Or `sly-connection-attempt-interval`

File: sly.el [hunk 1]
```diff
@@ -1395,6 +1395,14 @@ Fall back to `sly-init-using-slynk-loader' if ASDF fails."
 (defun sly-read-port-and-connect (inferior-process)
   (sly-attempt-connection inferior-process nil 1))
 
+(defcustom sly-connection-update-interval 0.3
+  "Interval at which the connection is polled when first connecting.
```
Comment (joaotavora) [inline sly.el:1399]: `Seconds to wait between connection attempts when first connecting.`

No more docstring verbiage is needed.

**Comment by Ambrevar** [comment]
Done again :)

**Comment by joaotavora** [comment]
Thanks!

**Pull request merged**